beesdoo_shift_swap/models/subscribed_underpopulated_shift.py

```diff
@@ -0,0 +1,282 @@
+from odoo import models, fields, api
+from datetime import datetime, timedelta, date
+from odoo.exceptions import Warning
+
+def daterange(start_date, end_date):
+    for n in range(int ((end_date - start_date).days)) :
+        return start_date + timedelta(n)
+
+
+class subscribe_underpopulated_shift(models.Model):
+    _name = 'beesdoo.shift.subscribed_underpopulated_shift'
+
+
+    def _get_selection_status(self):
+        return [
+            ('draft','Draft'),
+            ('validate','Validate'),
+            ('done','Done')
+        ]
+
+    state=fields.Selection(
+        selection=_get_selection_status,
+        default='draft'
+    )
+    worker_id = fields.Many2one(
+        "res.partner",
+        domain=[
+            ("is_worker", "=", True),
+            ("working_mode", "in", ("regular", "irregular")),
+            ("state", "not in", ("unsubscribed", "resigning")),
+        ],
+    )
+    exchanged_timeslot_id = fields.One2many(
+        comodel_name='beesdoo.shift.timeslots_date',
+        inverse_name='id',
+        string='exchanged_shift'
+    )
+    exchange_status = fields.Boolean(default=False, string="status exchange shift")
+    exchanged_shift_id = fields.One2many(
+        comodel_name='beesdoo.shift.shift',
+        inverse_name='id',
+        compute='is_shift_exchanged_already_generated'
+    )
+    comfirmed_timeslot_id = fields.One2many(
+        comodel_name='beesdoo.shift.timeslots_date',
+        inverse_name='id',
+        string='asked_shift'
+    )
+    comfirmed_shift_id = fields.One2many(
+        comodel_name='beesdoo.shift.shift',
+        inverse_name='id',
+        compute='is_shift_comfirmed_already_generated'
+    )
+    comfirme_status = fields.Boolean(default=False, string="status comfirme shift")
+    date = fields.Date(required=True)
+
+    def update_status(self):
+        if self.exchanged_timeslot_id and self.comfirmed_timeslot_id and self.worker_id and self.date:
+            self.write({"state" : "validate"})
+            if self.exchange_status and self.comfirme_status :
+                self.write({"state" : "done"})
+
+    @api.multi
+    def is_shift_exchanged_already_generated(self):
+        # Get current user
+        #cur_user = self.env["res.users"].browse(self.uid)
+
+        #if the supercooperateur make the exchange
+        cur_user = self.worker_id
```

Comment:
Pay attention, that here, you're using a `res.partner` and not a `res.users`. So your variable seams to be named wrongly.

Reply:
I renamed it current_worker